Link_PreferredBlocks.bas

```diff
@@ -47,6 +49,28 @@ For proCntr = 0 To nmbrPro
     'link preferredblocks to the professor. It's optional since every preferredblock has a professor
     professor.preferredBlocks = prefBlocks
 
+  'preferred course
+                'count number of courses
+nmbrCourse = Sheet1.Cells(1, Columns.Count).End(xlToLeft).Column
+
+For proCntr = 0 To nmbrPro
+    
+    'instantiate a preferred courses collection
+    Dim prefCourses As Collection
+    Set prefCrouses = New Collection
+    
+    'read preferred blocks
+    For nmbrCourse = 1 To numbrCourse
+        Set preferredCourse = New CPreferredCourse
+
+        
+        preferredCourse.PreferredBlockID = nmbrCourse
+        preferredCourse.PreferredLevel = ActiveCell.Offset(proCntr, (30 + nmbrBlock)).Value
+        preferredCourse.ProfessorName = professor.ProfessorName
+        
+        prefCourse.Add preferredCourse
+    

```

Comment:
There should be a 'next' for the inner for loop. Also, I suggest putting the preferred course down below the preferred block assignment, so that each preferred course is attached with the professor.